packages/design-system/src/components/primitives/numeric-gesture-control.ts

```diff
@@ -231,11 +236,30 @@ export const numericScrubControl = (
   };
 };
 
+const setRootStyle = (
```

Comment:
Imo such code should have guard protection against parallel exection i.e.
It must not be called from 2 different places with the same property.
```
setRooltStyle // A component
setRootStyle // B component
dispose // A
dispose // B
```
broke root style

2 ways to avoid.
Use per property counter and save/restore original value into module level variable only if counter is 0

Or just fail if counter[prop] > 1

Reply:
agree, its technically impossible to have twice as this is used based on a single user interaction, so if we look at it from how its used, it should be fine, but from purely API perspective - yeah in theory its possible. 

Do you see any way this is a real concern or are we bike shedding here?